src/staking_rewards/main.py

```diff
@@ -0,0 +1,101 @@
+
+
+def get_epoch_info():
+    staking_contract = web3.eth.contract(address=address_staking, abi=abi_staking)
+    try:
+        epoch_info = staking_contract.functions.epoch().call()
+    except ValueError:
+        epoch_info = None
+
+    return epoch_info
+
+
+def get_total_supply():
+    sklima_contract = web3.eth.contract(address=address_sklima, abi=abi_sklima)
+    try:
+        total_supply = sklima_contract.functions.totalSupply().call()
+        return total_supply
+    except Exception:
+        pass
+
+
+def get_rebase_info(rebase_number):
+    sklima_contract = web3.eth.contract(address=address_sklima, abi=abi_sklima)
+    try:
+        rebase = sklima_contract.functions.rebases(rebase_number).call()
+        return rebase
+    except Exception:
+        pass
+
+
+def get_info():
+    supply = get_total_supply()
+    index = supply / 5e15
+
+    epoch_info = get_epoch_info()
+    next_epoch_number = epoch_info[1]
+
+    # Note: can't think of a better way to figure out the current rebase number.
+    rebase_number = next_epoch_number - 3
+    rebase = get_rebase_info(rebase_number)
+
+    # Note: Not sure this is the best way of working out the reward rate, but it seems to give similar figures
+    staked_before = rebase[2]
+    staked_after = rebase[3]
+    diff_ratio = (abs(staked_after - staked_before) / staked_before) + 1
+    yearly_roi = ((diff_ratio ** (365 * 3)) - 1) * 100
+
+    return [index, yearly_roi]
+
+
+@client.event
+async def on_ready():
+    print('Logged in as {0.user}'.format(client))
+    if not update_info.is_running():
+        update_info.start()
+
+
+@tasks.loop(seconds=60)
+async def update_info():
+    index, yearly_roi = get_info()
+
+    print(f'Index: {index:,.2f} KLIMA')
+    print(f'APY: {yearly_roi:,.2f}%')
+
+    for guild in client.guilds:
+        guser = guild.get_member(client.user.id)
+        await guser.edit(nick=f'{yearly_roi:,.2f}% APY')
+
+        await client.change_presence(activity=discord.Activity(type=discord.ActivityType.playing,
```

Comment:
Can we break this line up a bit with some whitespace for clarity? Like here https://github.com/KlimaDAO/discord-bots/blob/main/src/next_rebase/main.py#L89

Reply:
Sure, updated as suggested in the latest commit.